plutus-use-cases/src/Plutus/Contracts/Uniswap/Data.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE NoImplicitPrelude          #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE RecordWildCards            #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# options_ghc -fno-warn-orphans          #-}
+{-# options_ghc -Wno-redundant-constraints #-}
+{-# options_ghc -fno-strictness            #-}
+{-# options_ghc -fno-specialise            #-}
+
+module Plutus.Contracts.Uniswap.Data
```

Comment:
I think it's traditional to name this sort of module "Types" rather than "Data" (although that's a perfectly reasonable name).